js/web/lib/wasm/binding/ort-wasm.d.ts

```diff
@@ -31,7 +31,23 @@ export interface OrtWasmModule extends EmscriptenModule {
   _OrtReleaseTensor(tensorHandle: number): void;
   _OrtRun(
       sessionHandle: number, inputNamesOffset: number, inputsOffset: number, inputCount: number,
-      outputNamesOffset: number, outputCount: number, outputsOffset: number): number;
+      outputNamesOffset: number, outputCount: number, outputsOffset: number, runOptionsHandle: number): number;
+
+  _OrtCreateSessionOptions(): number;
```

Comment:
It's a trade-off. I initially designed that way, but we can't leverage optional values with a single API, i.e. we must call every session options setting APIs even though it's not necessary.

Reply:
those APIs are very cheap in implementation ( most of them are just a struct member assignment of simple type ), and they can still avoid to be called.

for example, in api.cc, we can check if logid == nullptr we don't call set session log ID.

Reply:
I think the key point is, whether we have the requirement to call those functions separately. If we always call those functions **ONLY** after creating an option object (it looks like so), we can make them into one API; otherwise we can keep them

Reply:
We don't support a full set of SessionOptions and RunOptions here. When we support full, it may be an overhead at runtime. Also, ignoring an error code from low layer is not a good design. I suggest that we re-visit this issue when we plan to support full options.